src/control/common/storage/pb_types.go

```diff
@@ -130,6 +142,19 @@ func (ncs NvmeControllers) StringHealthStats() string {
 	return buf.String()
 }
 
+// Summary reports accumulated storage space and the number of controllers.
+func (ncs NvmeControllers) Summary() string {
```

Comment:
I'm really not sure I agree with setting this precedent of adding more UI-focused functionality to the types. I know I'm starting to sound like a broken record with this SOLID stuff, but let's step back a bit and look at what's going on in this PR:

We've got a net change of ~400LOC across 15 files in an effort to summarize storage output for the user. Any updates to the output format will likely need to touch a large subset of these files. Any changes to the storage types themselves will likely have a ripple effect across other files.

I really think we should be focusing on keeping the core types mostly "pure" and independent of any UI concerns. Stringers are useful for developer output and quick POC UIs, but they're not a good fit for finished systems.

I propose that the best way to approach this problem is to handle formatting and UI at the edge (i.e. on the outside of the client API). At the moment, this is just dmg. In the future, there could be other consumers of client API output, and those consumers will do their own slicing and dicing of the data.

Reply:
these types are to be moved to the edge when we re-factor client NVMe, then the Stringer implementations will live in the client package

Reply:
I don't think that the point of my comment was addressed. It's all well and good to say that we're going to redo it later, but the more tech debt we accumulate the harder it is to change things later. We are quickly running out of time to "do it right later"... Later is almost here. Soon we'll be moving on to another phase of the contract work and will have less ability to focus on this stuff.

Reply:
do you want a specific change or is it okay to revisit this when we stop using pb_types from within dmg?

Reply:
That's fine. We can tackle this later. I'm just sensitive to making more work for ourselves down the road with a bunch of small changes that add up to big painful upheavals later.